ddtrace/profiling/collector/_lock.py

```diff
@@ -233,6 +235,33 @@ def __enter__(self, *args, **kwargs):
     def __exit__(self, *args, **kwargs):
         self._release(self.__wrapped__.__exit__, *args, **kwargs)
 
+    # Get lock acquire/release call location and variable name the lock is assigned to
+    def _get_lock_call_loc_with_name(self) -> typing.Optional[str]:
+        try:
+            # We expect the call stack to be like this:
```

Comment:
Can we add a debug_assert here so we'll catch violations in test?

Reply:
I'm not sure whether Python or this repo has support for something similar to Rust's `std::debug_assert!`.

Reply:
We could mock it using a config variable?

Reply:
Oh that would work. I'll follow up with another PR.